dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/client/metadata/MetadataUtils.java

```diff
@@ -176,7 +176,7 @@ public static RemoteMetadataService referMetadataService(ServiceInstance instanc
         RemoteMetadataService remoteMetadataService;
         ProxyFactory proxyFactory =
                 applicationModel.getExtensionLoader(ProxyFactory.class).getAdaptiveExtension();
-        if (useV2) {
+        if (useV2 && !inNativeImage) {
```

Comment:
Why can't use v2 in native image here?

Reply:
MetadataServiceV2 only supports `nativestub` proxy mode. This needs to be supported separately, and the original problem should be fixed first in this pr.